### PR TITLE
Move observed containers test to calculator

### PR DESCRIPTION
**Pull request by bskiba**

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/observed_containers_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+			pod:           test.Pod().Get(),
+			expectedPatch: addVpaObservedContainersPatch([]string{}),
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			c := NewObservedContainersCalculator()
+			patches, err := c.CalculatePatches(tc.pod, nil)
+			assert.NoError(t, err)
+			if assert.Len(t, patches, 1, "Unexpected amount of patches.") {
```
Comment (krzysied) [inline vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/observed_containers_test.go]: nit: s/amount/number/

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/util.go [hunk 2]
```diff
@@ -39,3 +43,16 @@ func GetAddAnnotationPatch(annotationName, annotationValue string) resource_admi
...
+// EqPatch returns true if patches are equal by comparing their
+// marshalling result.
+func EqPatch(a, b resource_admission.PatchRecord) bool {
+	aJson, aErr := json.Marshal(a)
+	bJson, bErr := json.Marshal(b)
+	return string(aJson) == string(bJson) && aErr == bErr
+}
+
+// AssertEqPatch asserts patches are equal.
+func AssertEqPatch(t *testing.T, got, want resource_admission.PatchRecord) {
```
Comment (krzysied) [inline vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/util.go]: nit: When golang package is vendored, the _test.go are omitted. IMHO it is good to keep code referencing "testing" separated from main code. I'm not sure what is golang official best practice for this case, though.

Comment (krzysied) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/util.go]: If we consider keeping "AssertEqPatch" in util, maybe we should also have "assertPatchOneOf" there?

Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/util.go]: Moved to test_utils file. I will move assertPatchOneOf with the next PR (resource updates calculator test)

File: vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/observed_containers_test.go (hunk 1 above)
Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/resource/pod/patch/observed_containers_test.go]: Done

**Pull request merged**